core/trino-main/src/main/java/io/trino/tracing/TracingAccessControl.java

```diff
@@ -0,0 +1,742 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tracing;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.trino.metadata.QualifiedObjectName;
+import io.trino.security.AccessControl;
+import io.trino.security.SecurityContext;
+import io.trino.spi.connector.CatalogSchemaName;
+import io.trino.spi.connector.CatalogSchemaTableName;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.function.FunctionKind;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.Privilege;
+import io.trino.spi.security.TrinoPrincipal;
+import io.trino.spi.security.ViewExpression;
+import io.trino.spi.type.Type;
+
+import java.security.Principal;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static io.trino.tracing.ScopedSpan.scopedSpan;
+import static java.util.Objects.requireNonNull;
+
+public class TracingAccessControl
+        implements AccessControl
+{
+    private final Tracer tracer;
+    private final AccessControl delegate;
+
+    @Inject
+    public TracingAccessControl(Tracer tracer, @ForTracing AccessControl delegate)
+    {
+        this.delegate = requireNonNull(delegate, "delegate is null");
+        this.tracer = requireNonNull(tracer, "tracer is null");
+    }
+
+    @VisibleForTesting
+    public AccessControl getDelegate()
+    {
+        return delegate;
+    }
+
+    @Override
+    public void checkCanSetUser(Optional<Principal> principal, String userName)
+    {
+        Span span = startSpan("checkCanSetUser");
+        try (var ignored = scopedSpan(span)) {
```

Comment:
nit: can we instead introduce `scopedStartedSpan(String name)` so this code becomes:

`try (var ignored = scopedStartedSpan("checkCanSetUser")) {`

Reply:
i copied this from TracingMetadata.
i guess the benefit of current approach is that we can stuck additional information into the scope easier.